### PR TITLE
Fix #1792:Fixed change in keyboard when show password was pressed

### LINKED ISSUE #1792: Toggling show/hide changes soft keyboard
**Describe the bug**
When we are on the screen where we have to input admin pin, we have a show and hide button, toggling which changes the soft keyboard 

**To Reproduce**
Steps to reproduce the behaviour:
1. Install app -> click on the admin profile
2. click on the show -> hide -> soft keyboard changes 

**Expected behaviour**
The soft keyboard should not change

**Video**
https://lh3.googleusercontent.com/2ACs5tLuu3JmS1V248M2flkFCL35GeI4ZaHaj2yXu-AIaXExQFbWXVDkkyeS8vCXBKIyU5RRxhQeJ08=m37

**Device**
- Redmi Note 5 Pro

### PR DESCRIPTION
**Pull request by Victor-Titan**
## Explanation

Fixes #1792 where the soft keyboard changes on pressing the show/hide button.

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.
ezgif com-video-to-gif (1)

### CODE REVIEW

**Comment by Victor-Titan** [comment]
PTAL @anandwana001

File: app/src/main/java/org/oppia/android/app/profile/PinPasswordActivityPresenter.kt
Comment (anandwana001) [inline app/src/main/java/org/oppia/android/app/profile/PinPasswordActivityPresenter.kt]: try to do it with data binding, in **layout file only** by setting `showPassword` to this newly introduced passwordHidden method.

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks @Victor-Titan 
This PR is on right track, works perfectly correct. Added few changes. 
Also, update the PR title as per given in the checklist of PR description.

File: app/src/main/java/org/oppia/android/app/profile/PinPasswordActivityPresenter.kt
Comment (Victor-Titan) [reply app/src/main/java/org/oppia/android/app/profile/PinPasswordActivityPresenter.kt]: I don't get you, isn't the value in the pinViewModel.showPassword already retrieved from data binding in the xml file?

Comment (anandwana001) [reply app/src/main/java/org/oppia/android/app/profile/PinPasswordActivityPresenter.kt]: Please refer this reference - https://developer.android.com/topic/libraries/data-binding/expressions

**Comment by Victor-Titan** [comment]
@anandwana001 please check now

File: app/src/main/java/org/oppia/android/app/profile/PinPasswordActivityPresenter.kt
Comment (anandwana001) [reply app/src/main/java/org/oppia/android/app/profile/PinPasswordActivityPresenter.kt]: @Victor-Titan Please don't resolve these comments, the author needs to resolve such comments.

**Comment by anandwana001** [comment]
> @anandwana001 please check now

@Victor-Titan  
It's not what I am looking for. We could use Data Binding where you only had to update or add stuff in the layout file. No kotlin work required. 
Reference - https://developer.android.com/topic/libraries/data-binding/expressions

File: app/src/main/java/org/oppia/android/app/profile/PinPasswordActivityPresenter.kt
Comment (Victor-Titan) [reply app/src/main/java/org/oppia/android/app/profile/PinPasswordActivityPresenter.kt]: @anandwana001 So are you asking me to invoke the method directly inside the XML file and not when in the onClickListener block?

Comment (anandwana001) [reply app/src/main/java/org/oppia/android/app/profile/PinPasswordActivityPresenter.kt]: Yes, not kotlin work, just the data binding in xml file will work.

Comment (Victor-Titan) [reply app/src/main/java/org/oppia/android/app/profile/PinPasswordActivityPresenter.kt]: I can't call that method in the xml file, please help

Comment (anandwana001) [reply app/src/main/java/org/oppia/android/app/profile/PinPasswordActivityPresenter.kt]: `app:passwordHidden="@{viewmodel.showPassword}"` That's all.

Comment (Victor-Titan) [reply app/src/main/java/org/oppia/android/app/profile/PinPasswordActivityPresenter.kt]: I didn't know there was such an attribute for the pinView, thank you.

File: app/src/main/res/layout/pin_password_activity.xml
Comment (anandwana001) [inline app/src/main/res/layout/pin_password_activity.xml]: please correct the formatting. You can use reformat code from the menu.

Comment (anandwana001) [reply app/src/main/res/layout/pin_password_activity.xml]: Also, check this setup for future work. https://github.com/oppia/oppia-android/wiki/Ktlint-Guide#macros

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Currently, the implementation seems opposite. Please check the applied condition.

Screenshot_1603590524

**Review by anandwana001 — APPROVED** [review]
LGTM, Thanks @Victor-Titan

**Review by BenHenning — APPROVED** [review]
Thanks @Victor-Titan! Approval for code owners.

Also, is it possible to add tests for this change to guard against regressions? E.g. maybe checking that the correct properties on the PinView are set at the expected times?

/cc @anandwana001 for checking for tests.

**Comment by Victor-Titan** [comment]
> Thanks @Victor-Titan! Approval for code owners.
> 
> Also, is it possible to add tests for this change to guard against regressions? E.g. maybe checking that the correct properties on the PinView are set at the expected times?
> 
> /cc @anandwana001 for checking for tests.

I'm sorry, but I have no idea how to write tests for code, much less for XMLs in Android, @anandwana001 @BenHenning can you please refer some material?

**Comment by anandwana001** [comment]
@Victor-Titan 
Please refere to the `PinPasswordActivityTest.kt`

**Comment by Victor-Titan** [comment]
I don't know how to access the keyboard input values via data binding for the tests, @anandwana001 @BenHenning please help?

**Comment by anandwana001** [comment]
> I don't know how to access the keyboard input values via data binding for the tests, @anandwana001 @BenHenning please help?

Did you check my earlier comment? - https://github.com/oppia/oppia-android/pull/2029#issuecomment-717191281

**Comment by Victor-Titan** [comment]
@anandwana001 I have tests and assignments due and won't be able to write the tests anytime soon for this PR, can we merge this and create another issue for adding tests?

**Comment by anandwana001** [comment]
@Victor-Titan we cannot merge without test case, as test case is something which says the concrete correctness of any fix. You can mark this PR draft, and work again once you come back from your assignments.

**Comment by anandwana001** [comment]
@Victor-Titan any update on this?

**Comment by anandwana001** [comment]
Closing this PR as no further update.

**Pull request closed without merging**
